x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -0,0 +1,386 @@
+   */
+  private indexInitialization?: Promise<void>;
+
+  constructor(options: Readonly<SessionIndexOptions>) {
+    this.#options = options;
+    this.#idleIndexCleanupTimeout = this.#options.config.session.idleTimeout
+      ? this.#options.config.session.idleTimeout.asMilliseconds() * 3
+      : null;
+  }
+
+  /**
+   * Retrieves session value with the specified ID from the index. If session value isn't found
+   * `null` will be returned.
+   * @param sid Session ID.
+   */
+  async get(sid: string) {
+    try {
+      const response = await this.#options.clusterClient.callAsInternalUser('get', {
+        id: sid,
+        ignore: [404],
+        index: SESSION_INDEX_ALIAS,
+      });
+
+      const docNotFound = response.found === false;
+      const indexNotFound = response.status === 404;
+      if (docNotFound || indexNotFound) {
+        this.#options.logger.debug('Cannot find session value with the specified ID.');
+        return null;
+      }
+
+      return {
+        sid,
+        ...response._source,
+        metadata: { primaryTerm: response._primary_term, sequenceNumber: response._seq_no },
+      } as Readonly<SessionIndexValue>;
+    } catch (err) {
+      this.#options.logger.error(`Failed to retrieve session value: ${err.message}`);
+      throw err;
+    }
+  }
+
+  /**
+   * Creates a new document for the specified session value.
+   * @param sessionValue Session index value.
+   */
+  async create(sessionValue: Readonly<Omit<SessionIndexValue, 'metadata'>>) {
+    if (this.indexInitialization) {
+      this.#options.logger.warn(
+        'Attempted to create a new session while session index is initializing.'
+      );
+      await this.indexInitialization;
+    }
+
+    const { sid, ...sessionValueToStore } = sessionValue;
+    try {
+      const {
+        _primary_term: primaryTerm,
+        _seq_no: sequenceNumber,
+      } = await this.#options.clusterClient.callAsInternalUser('create', {
+        id: sid,
+        // We cannot control whether index is created automatically during this operation or not.
+        // But we can reduce probability of getting into a weird state when session is being created
+        // while session index is missing for some reason. This way we'll recreate index with a
+        // proper name and alias. But this will only work if we still have a proper index template.
+        index: SESSION_INDEX_NAME,
+        body: sessionValueToStore,
+        refresh: 'wait_for',
+      });
+
+      return { ...sessionValue, metadata: { primaryTerm, sequenceNumber } } as SessionIndexValue;
+    } catch (err) {
+      this.#options.logger.error(`Failed to create session value: ${err.message}`);
+      throw err;
+    }
+  }
+
+  /**
+   * Re-indexes updated session value.
+   * @param sessionValue Session index value.
+   */
+  async update(sessionValue: Readonly<SessionIndexValue>) {
+    const { sid, metadata, ...sessionValueToStore } = sessionValue;
```

Comment:
As I mentioned I was doing this temporary workaround (until we get proper plugin status support from Core) mostly to please our tests and because `create` can create index before we create a proper template for it. Neither of this  seems to be a problem for `update`, that's why I didn't add it here. It'd wouldn't hurt to add it here though if you think it makes core more consistent?

Reply:
I did some testing, and I agree with you. I was worried that deleting the index just prior to updating a session would be problematic, but that doesn't appear to be the case.